.github/workflows/integration.yaml

```
@@ -370,3 +370,172 @@ jobs:
+          # Build all Go std test binaries from gotip
+          pushd $(gotip env GOROOT)
+          for value in src/archive/tar \
+            src/bufio \
+            src/bytes \
+            src/context \
+            src/encoding/ascii85 \
+            src/encoding/asn1 \
+            src/encoding/base32 \
+            src/encoding/base64 \
+            src/encoding/binary \
+            src/encoding/csv \
+            src/encoding/gob \
+            src/encoding/hex \
+            src/encoding/json \
+            src/encoding/pem \
+            src/encoding/xml \
+            src/errors \
+            src/expvar \
+            src/flag \
+            src/fmt \
+            src/hash \
+            src/hash/adler32 \
+            src/hash/crc32 \
+            src/hash/crc64 \
+            src/hash/fnv \
+            src/hash/maphash \
+            src/io \
+            src/io/fs \
+            src/io/ioutil \
+            src/log \
+            src/log/syslog \
+            src/maps \
+            src/math \
+            src/math/big \
+            src/math/bits \
+            src/math/cmplx \
+            src/math/rand \
+            src/mime \
+            src/mime/multipart \
+            src/mime/quotedprintable \
+            src/os \
+            src/os/exec \
+            src/os/signal \
+            src/os/user \
+            src/path \
+            src/path/filepath \
+            src/reflect \
+            src/regexp \
+            src/regexp/syntax \
+            src/runtime \
+            src/runtime/internal/atomic \
+            src/runtime/internal/math \
+            src/runtime/internal/sys \
+            src/slices \
+            src/sort \
+            src/strconv \
+            src/strings \
+            src/sync \
+            src/sync/atomic \
+            src/syscall \
+            src/testing \
+            src/testing/fstest \
+            src/testing/iotest \
+            src/testing/quick \
+            src/time
+          do
+            GOOS=wasip1 GOARCH=wasm gotip test -v -c -o ${{ env.BINARY_OUT }}/tests/${value//\//_}.test ./$value
+          done
+          # Copy src from gotip to run tests against.
+          mv src ${{ env.BINARY_OUT }}
+
+  go_tests:
+    needs: build_go_tests
+    name: Go (${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false # don't fail fast as sometimes failures are arch/OS specific
+      matrix:
+        os: [ubuntu-22.04, macos-12, windows-2022]
+    env:
+      BINARY_OUT: ~/go-test-binaries
+
+    steps:
+      - name: Cache Go test binaries
+        id: cache-go-test-binaries
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ${{ env.BINARY_OUT }}
+          key: go-test-binaries-${{ env.GO_VERSION }}
+
+      - if: ${{ steps.cache-go-test-binaries.outputs.cache-hit != 'true' }}
+        run: |
+          echo "Go test binaries cache is empty"
+          exit 1
```

let's use `fail-on-cache-miss` like other jobs
```suggestion
        with:
          path: ${{ env.BINARY_OUT }}
          key: go-test-binaries-${{ env.GO_VERSION }}
          fail-on-cache-miss: true
```